packages/core/lib/use-puck.ts

```diff
@@ -16,9 +16,13 @@ export const usePuck = () => {
     appState,
     config,
     dispatch,
-    getPermissions: (selectedItem?: ComponentData) => {
+    getPermissions: (
+      selectedItem?: ComponentData,
+      type?: keyof DefaultComponentProps
+    ) => {
```

Comment:
We might wanna tweak this API a bit to so users aren't having to do `getPermissions(null, "Hero")`, but I think let's do that as part of the documentation to keep things moving.

Reply:
@chrisvxd I can move to an object which would help?

```tsx
getPermissions: ({
      selectedItem,
      type,
    }: {
      selectedItem?: ComponentData;
      type?: keyof DefaultComponentProps;
    }) => {
      return getPermissions({
        selectedItem: selectedItem || currentItem,
        type,
        globalPermissions: globalPermissions || {},
        config,
      });
    },
```